schema/fields.go

```diff
@@ -299,3 +346,18 @@ func (f *Field) IsPrimaryKey() bool {
 func (f *Field) IsAutoGenerated() bool {
 	return f.AutoGenerated != nil && *f.AutoGenerated
 }
+
+func (f *Field) IsCompatible(f1 *Field) error {
+	if f.DataType != f1.DataType {
+		return api.Errorf(api.Code_INVALID_ARGUMENT, "data type mismatch for field %q", f.FieldName)
+	}
+
+	if f.IsPrimaryKey() && !f1.IsPrimaryKey() {
```

Comment:
`f.IsPrimaryKey() != f1.IsPrimaryKey()` ?

Reply:
ah, right - that will work.